src/MapViewController.swift

```diff
@@ -169,12 +256,16 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
   open func loadSceneFile(_ path: String, sceneUpdates: [TGSceneUpdate]) {
     tgViewController.loadSceneFile(path, sceneUpdates: sceneUpdates)
   }
-  
-  open func loadSceneFileAsync(_ path: String) {
+
+  open func loadSceneFileAsync(_ path: String, onSceneLoaded: OnSceneLoaded?) {
+    self.onSceneLoaded = onSceneLoaded
+    tgViewController.mapViewDelegate = mapViewDelegatesExist() ? self : nil
```

Comment:
Here and in the similar function below, we would only care really if onSceneLoaded is non-nil, so we could just check for that?